src/async_jobs.ts

```diff
@@ -31,46 +36,97 @@ async function slackInteractivityHandler(payload: SlackEventPayload) {
+    const redisData = await RedisApiUtil.getHash(redisClient, redisHashKey);
 
-  const selectedVoterStatus = payload.actions[0].selected_option
-    ? payload.actions[0].selected_option.value
-    : payload.actions[0].value;
-  if (selectedVoterStatus) {
-    logger.info(
-      `SERVER POST /slack-interactivity: Determined user interaction is a voter status update or undo.`
-    );
-    await SlackInteractionHandler.handleVoterStatusUpdate({
-      payload,
-      selectedVoterStatus,
-      originatingSlackUserName,
-      slackChannelName: originatingSlackChannelName,
-      userPhoneNumber: redisData.userPhoneNumber,
-      twilioPhoneNumber: redisData.twilioPhoneNumber,
-      redisClient,
-    });
-  } else if (payload.actions[0].selected_user) {
-    logger.info(
-      `SERVER POST /slack-interactivity: Determined user interaction is a volunteer update.`
-    );
-    await SlackInteractionHandler.handleVolunteerUpdate({
-      payload,
-      originatingSlackUserName,
-      slackChannelName: originatingSlackChannelName,
-      userPhoneNumber: redisData.userPhoneNumber,
-      twilioPhoneNumber: redisData.twilioPhoneNumber,
-    });
+    if (!redisData) {
+      logger.debug(
+        `SERVER POST /slack-interactivity: Received an interaction for a voter who no longer exists in Redis.`
+      );
+      return;
+    }
+
+    switch (payload.type) {
+      case 'block_actions': {
+        const selectedVoterStatus = payload.actions[0].selected_option
+          ? payload.actions[0].selected_option.value
+          : payload.actions[0].value;
+        if (selectedVoterStatus) {
+          logger.info(
+            `SERVER POST /slack-interactivity: Determined user interaction is a voter status update or undo.`
+          );
+          await SlackInteractionHandler.handleVoterStatusUpdate({
+            payload,
+            selectedVoterStatus,
+            originatingSlackUserName,
+            slackChannelName: originatingSlackChannelName,
+            userPhoneNumber: redisData ? redisData.userPhoneNumber : null,
+            twilioPhoneNumber: redisData ? redisData.twilioPhoneNumber : null,
+            redisClient,
+          });
+        } else if (payload.actions[0].selected_user) {
+          logger.info(
+            `SERVER POST /slack-interactivity: Determined user interaction is a volunteer update.`
+          );
+          await SlackInteractionHandler.handleVolunteerUpdate({
+            payload,
+            originatingSlackUserName,
+            slackChannelName: originatingSlackChannelName,
+            userPhoneNumber: redisData ? redisData.userPhoneNumber : null,
+            twilioPhoneNumber: redisData ? redisData.twilioPhoneNumber : null,
+          });
+        }
+        return;
+      }
+      case 'message_action': {
+        logger.info(
+          `SERVER POST /slack-interactivity: Determined user interaction is a message shortcut.`
+        );
+        if (payload.callback_id === 'reset_demo') {
```

Comment:
Nit: In https://github.com/tomerovadia/VoterHelpLine/pull/84, I created a new module + enum for these `callback_id`. It'll be nice to consolidate these in one place.

Reply:
That's awesome, thanks for doing that! I'm about to hop over to look at all the changes you sent.